homeassistant/components/broadlink/helpers.py

```diff
@@ -47,3 +50,25 @@ def import_device(hass, host):
             data={CONF_HOST: host},
         )
         hass.async_create_task(task)
+
+
+def get_ip_or_none(host):
+    """Return an IP address or None if socket.EAI_NONAME."""
+    try:
+        return socket.gethostbyname(host)
+    except OSError:
+        return None
+
+
+def get_broadcast_addrs():
+    """Return all available IPv4/L3 broadcast addresses."""
+    return [
+        snic.broadcast
+        for _, snics in psutil.net_if_addrs().items()
```

Comment:
It's questionable to add this to the integration code since we want transport and protocol code to be in a separate library. Preferably the library should provide this scan.